File: FournilBack/src/main/java/fr/fournil/bakery/model/entities/ProductInCart.java
package fr.fournil.bakery.model.entities;

import java.sql.Date;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.validation.constraints.Min;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "ProductInCart")
public class ProductInCart {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	// @JsonIgnoreProperties(value={"commande", "photoProductSet",
	// "productComposition",
	// "productPerKgPrice","productSpecification","productSummary","limitStockProduct"
	// })

	@OneToOne(cascade = { CascadeType.PERSIST }, fetch = FetchType.EAGER)
	// @JoinColumn(name = "product_id")
	private Product product;

	@Min(1)
	private Integer quantity;
	
	@OneToOne(cascade = { CascadeType.PERSIST }, fetch = FetchType.EAGER)
	private Format format;

	@ManyToOne(cascade = { CascadeType.PERSIST }, fetch = FetchType.EAGER)
	private FrequencyDeliveryType frequencyDeliveryType;

	private double totalPricePerProduct;

	@ManyToMany(cascade = { CascadeType.PERSIST }, fetch = FetchType.EAGER)
	// @JoinTable(name = " productInCartListInShoppinCard", joinColumns =
	// @JoinColumn(name = "productInCart_id"), inverseJoinColumns = @JoinColumn(name
	// = "shoppingCart_id"))
	// @JoinColumn(name = "shoppingCart_id")
	
	private Set<ShoppingCart> shoppingCart = new HashSet<>();

	public ProductInCart() {
		super();
	}


	public ProductInCart(Long id, Product product, @Min(1) Integer quantity, Format format,
			FrequencyDeliveryType frequencyDeliveryType, double totalPricePerProduct, Set<ShoppingCart> shoppingCart) {
		super();
		this.id = product.getId();
		this.product = product;
		this.quantity = quantity;
		this.format = format;
		this.frequencyDeliveryType = frequencyDeliveryType;
		this.totalPricePerProduct =  (product.getProductPerKgPrice())*(format.getFormatWeight()*getQuantity())*frequencyDeliveryType.getFactorFrequency();
		this.shoppingCart = shoppingCart;
	}



	public ProductInCart(Product product, @Min(1) Integer quantity, Format format,
			FrequencyDeliveryType frequencyDeliveryType, double totalPricePerProduct, Set<ShoppingCart> shoppingCart) {
		super();
		this.product = product;
		this.quantity = quantity;
		this.format = format;
		this.frequencyDeliveryType = frequencyDeliveryType;
		this.totalPricePerProduct = (product.getProductPerKgPrice())*(format.getFormatWeight()*getQuantity())*(frequencyDeliveryType.getFactorFrequency());
		this.shoppingCart = shoppingCart;
	}


	public Long getId() {
		return product.getId();
	}


	public void setId(Long id) {
		this.id = id;
	}


	public Product getProduct() {
		return product;
	}


	public void setProduct(Product product) {
		this.product = product;
	}


	public Integer getQuantity() {
		return quantity;
	}


	public void setQuantity(Integer quantity) {
		this.quantity = quantity;
	}


	public Format getFormat() {
		return format;
	}


	public void setFormat(Format format) {
		this.format = format;
	}


	public FrequencyDeliveryType getFrequencyDeliveryType() {
		return frequencyDeliveryType;
	}


	public void setFrequencyDeliveryType(FrequencyDeliveryType frequencyDeliveryType) {
		this.frequencyDeliveryType = frequencyDeliveryType;
	}

	@Transient
	public double getTotalPricePerProduct() {
		return (getProduct().getProductPerKgPrice())*(getFormat().getFormatWeight()*getQuantity())*(getFrequencyDeliveryType().getFactorFrequency());
	}


	public void setTotalPricePerProduct(double totalPricePerProduct) {
		this.totalPricePerProduct = totalPricePerProduct;
	}


	public Set<ShoppingCart> getShoppingCart() {
		return shoppingCart;
	}


	public void setShoppingCart(Set<ShoppingCart> shoppingCart) {
		this.shoppingCart = shoppingCart;
	}


	@Override
	public int hashCode() {
		return Objects.hash(id, product);
	}




	@Override
	public String toString() {
		return "ProductInCart [id=" + id + ", product=" + product + ", quantity=" + quantity+"Format="+ format
				+ ", frequencyDeliveryType=" + frequencyDeliveryType + ", totalPricePerProduct=" + totalPricePerProduct
				+ ", shoppingCart=" + shoppingCart + "]";
	}

	
}

File: FournilFront/src/app/models/product-in-cart.ts
import { Format } from './format';
import { Frequence } from './frequence';
import { Product } from './product';

export class ProductInCart {

    private _id: number;
    private _product: Product;
    private _quantity: number;
    private _format: Format;
    private _frequencyDeliveryType: Frequence; 
    private _totalPricePerProduct: number;




	constructor(id: number, product: Product, quantity: number, format: Format, frequencyDeliveryType: Frequence, totalPricePerProduct: number) {
		this._id = id;
		this._product = product;
		this._quantity = quantity;
		this._format = format;
		this._frequencyDeliveryType = frequencyDeliveryType;
		this._totalPricePerProduct = totalPricePerProduct;
	}

    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter product
     * @return {Product}
     */
	public get product(): Product {
		return this._product;
	}

    /**
     * Getter quantity
     * @return {number}
     */
	public get quantity(): number {
		return this._quantity;
	}

    /**
     * Getter format
     * @return {Format}
     */
	public get format(): Format {
		return this._format;
	}

    /**
     * Getter frequencyDeliveryType
     * @return {Frequence}
     */
	public get frequencyDeliveryType(): Frequence {
		return this._frequencyDeliveryType;
	}

    /**
     * Getter totalPricePerProduct
     * @return {number}
     */
	public get totalPricePerProduct(): number {
		return this._totalPricePerProduct;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter product
     * @param {Product} value
     */
	public set product(value: Product) {
		this._product = value;
	}

    /**
     * Setter quantity
     * @param {number} value
     */
	public set quantity(value: number) {
		this._quantity = value;
	}

    /**
     * Setter format
     * @param {Format} value
     */
	public set format(value: Format) {
		this._format = value;
	}

    /**
     * Setter frequencyDeliveryType
     * @param {Frequence} value
     */
	public set frequencyDeliveryType(value: Frequence) {
		this._frequencyDeliveryType = value;
	}

    /**
     * Setter totalPricePerProduct
     * @param {number} value
     */
	public set totalPricePerProduct(value: number) {
		this._totalPricePerProduct = value;
	}
	


}

